skylark/skylark.bzl

```diff
@@ -73,93 +71,9 @@ def _skylark_doc_impl(ctx):
       progress_message = ("Generating Skylark doc for %s (%d files)"
                           % (ctx.label.name, len(sources))))
 
-_skylark_common_attrs = {
-    "srcs": attr.label_list(allow_files = _SKYLARK_FILETYPE),
-    "deps": attr.label_list(providers = ["transitive_bzl_files"],
-                            allow_files = False),
-}
-
-skylark_library = rule(
-    _skylark_library_impl,
-    attrs = _skylark_common_attrs,
-)
-"""Creates a logical collection of Skylark .bzl files.
-
-Args:
-  srcs: List of `.bzl` files that are processed to create this target.
-  deps: List of other `skylark_library` targets that are required by the Skylark
-    files listed in `srcs`.
-
-Example:
-  If you would like to generate documentation for multiple .bzl files in various
-  packages in your workspace, you can use the `skylark_library` rule to create
-  logical collections of Skylark sources and add a single `skylark_doc` target for
-  building documentation for all of the rule sets.
-
-  Suppose your project has the following structure:
-
-  ```
-  [workspace]/
-      WORKSPACE
-      BUILD
-      checkstyle/
-          BUILD
-          checkstyle.bzl
-      lua/
-          BUILD
-          lua.bzl
-          luarocks.bzl
-  ```
-
-  In this case, you can have `skylark_library` targets in `checkstyle/BUILD` and
-  `lua/BUILD`:
-
-  `checkstyle/BUILD`:
-
-  ```python
-  load("@io_bazel_skydoc//skylark:skylark.bzl", "skylark_library")
-
-  skylark_library(
-      name = "checkstyle-rules",
-      srcs = ["checkstyle.bzl"],
-  )
-  ```
-
-  `lua/BUILD`:
-
-  ```python
-  load("@io_bazel_skydoc//skylark:skylark.bzl", "skylark_library")
-
-  skylark_library(
-      name = "lua-rules",
-      srcs = [
-          "lua.bzl",
-          "luarocks.bzl",
-      ],
-  )
-  ```
-
-  To build documentation for all the above `.bzl` files at once:
-
-  `BUILD`:
-
-  ```python
-  load("@io_bazel_skydoc//skylark:skylark.bzl", "skylark_doc")
-
-  skylark_doc(
-      name = "docs",
-      deps = [
-          "//checkstyle:checkstyle-rules",
-          "//lua:lua-rules",
-      ],
-  )
-  ```
-
-  Running `bazel build //:docs` would build a single zip containing documentation
-  for all the `.bzl` files contained in the two `skylark_library` targets.
-"""
-
 _skylark_doc_attrs = {
+    "srcs": attr.label_list(allow_files = _SKYLARK_FILETYPE),
+    "deps": attr.label_list(allow_files = False),
```

Comment:
This should have "providers = [SkylarkLibraryInfo]", right?

Reply:
My only concern was that current skylark_doc targets might have deps that aren't skylark_library targets, but I guess they all had to be previously. I'll fix.